crates/wal-protocol/src/lib.rs

```diff
@@ -43,74 +43,18 @@ impl Envelope {
         Self { header, command }
     }
 
-    pub fn encode(&self) -> Result<Bytes, SerializationError> {
+    pub fn to_bytes(&self) -> Result<Bytes, StorageSerdeError> {
         let mut buf = BytesMut::default();
-        EnvelopeSerializer::serialize(self, &mut buf)?;
+        self.encode(&mut buf)?;
         Ok(buf.freeze())
     }
 
-    pub fn decode(bytes: impl AsRef<[u8]>) -> Result<Self, SerializationError> {
-        EnvelopeSerializer::deserialize(bytes.as_ref())
+    pub fn from_bytes(bytes: impl AsRef<[u8]>) -> Result<Self, StorageSerdeError> {
+        Self::decode(bytes.as_ref())
```

Comment:
Would the be replaced directly by decode in a follow up? I'm not sure if to_bytes and from_bytes would still be needed.

Reply:
`from_bytes` can directly be replaced yes. `to_bytes` is a bit of a convenience method which also creates the buffer into which the envelope gets encoded.